README
Bundle/Bricolage version 2.0.0
==============================

DESCRIPTION

The bundle provides an easy way to install all of the modules required by
Bricolage. Bricolage is a 100% pure Perl content-management and publishing
system which features intuitive and highly-configurable administration,
workflow, permissions, templating, server-neutral output, distribution, and
document management. Its home page is http://bricolagecms.org/.

INCLUDED MODULES

Storable                 - Persistency for perl data structures
Time::HiRes              - High resolution ualarm, usleep, and gettimeofday
Unix::Syslog             - Perl interface to the UNIX syslog(3) calls
Net::Cmd                 - Network Command class (as used by FTP, SMTP etc)
Devel::Symdump           - Dump symbol names or the symbol table
DBI 1.18                 - Database independent interface for Perl
Error                    - Error/exception handling in an OO-ish way
Cache::Cache             - The Cache interface
Cache::Mmap              - Shared data cache using memory mapped files
Digest::MD5              - Perl interface to the MD5 Algorithm
Digest::SHA1 2.01        - Perl interface to the SHA-1 Algorithm
URI                      - Uniform Resource Identifiers (absolute and relative)
HTML::Tagset             - Data tables useful in parsing HTML
HTML::Parser             - HTML parser class
MIME::Base64             - Encoding and decoding of base64 strings
MIME::Tools              - Modules for parsing (and creating!) MIME entities
Mail::Address            - Parse mail addresses
XML::Writer              - Perl extension for writing XML documents
LWP                      - Library for WWW access in Perl
Image::Info              - Extract meta information from image files
MLDBM                    - Store multi-level hash structure in single level
                           tied hash
Params::Validate 0.57    - Validate method/function parameters
Exception::Class 1.12    - Perl Exceptions Base Class
Class::Container 0.09    - Glues object frameworks together transparently
mod_perl 1.30            - Embed a Perl interpreter in the Apache HTTP server
Apache::Request 1.0      - Generate compiler and linker flags for libapreq
HTML::Mason 1.23         - High-Performance, Dynamic Web Site Authoring System
DBD::Pg 1.22             - PostgreSQL database driver for the DBI module
DB_File                  - Perl5 access to Berkeley DB version 1.x
Apache::Session 1.54     - A persistence framework for session data
Test::Harness 2.03       - Run perl standard test scripts with statistics
Test::Simple             - Basic utilities for writing tests
Test::MockModule 0.04    - Override subroutines in a module for unit testing
Test::File::Contents 0.02  - Test routines for examining the contents of files
Text::Balanced           - Extract delimited text sequences from strings
XML::Parser 2.34         - A Perl module for parsing XML documents
XML::Simple              - Easy API to read/write XML (esp config files)
IO::Stringy              - I/O on in-core objects like strings and arrays
SOAP::Lite 0.55          - Client and server side SOAP implementation
File::Temp               - Return name and handle of a temporary file safely
Text::LevenshteinXS      - XS implementation of the Levenshtein edit distance
Locale::Maketext         - Framework for Localization in Perl
Test::Class 0.04         - xUnit/JUnit style Test Suite System
Params::CallbackRequest 1.16       - Functional and object-oriented callback
                                     architecture
MasonX::Interp::WithCallbacks 1.10 - Mason callback support via
                                     Params::CallbackRequest
Safe 2.09                - Compile and execute code in restricted compartments
DateTime 0.21            - Date and time objects
DateTime::TimeZone 0.2601  - Time zone object base class and factory
Term::ReadPassword       - Prompt for passwords without echoing to the terminal
Data::UUID               - Generate globally/universally unique identifiers
                           (GUIDs/UUIDs)
List::Util               - A selection of general-utility list subroutines
List::MoreUtils          - Provides the stuff missing in List::Util
Text::Diff               - Perform diffs on files and record sets
Text::Diff::HTML 0.02    - XHTML format for Text::Diff::Unified
Text::WordDiff           - Track changes between documents
URI::Escape              - Escape and unescape unsafe URI characters
Scalar::Util             - A selection of general-utility scalar subroutines

INSTALLATION

To install this module type the following as root:

   perl -MCPAN -e 'install Bundle::Bricolage'

COPYRIGHT AND LICENCE

Copyright (c) 2002-2008, David E. Wheeler. Some Rights Reserved.

This module is free software; you can redistribute it and/or modify it under
the same terms as Perl itself.



Changes
Revision history for Perl module Bundle::Bricolage.

2.0.0
      - Fixed encoding issue in Pod.
      - Updated URLs to point to bricolagecms.org.

1.11.0 2008-10-05T22:20:24
       - Added new modules required for Bricolage 1.11.1:
         + mod_perl 1.30
         + Test::File::Contents 0.02
         + Test::Class 0.04
         + Params::CallbackRequest 1.16
         + DateTime 0.21
         + DateTime::TimeZone 0.2601
         + Text::Diff
         + Text::Diff::HTML 0.02
         + Text::WordDiff
         + Scalar::Util
       - Removed Bundle::libnet.

1.10.0  2006-02-05T03:20:08
      - Added new modules required for Bricolage 1.9.1:
        + List::Util
        + List::MoreUtils
        + Text::LevenshteinXS
      - Removed Text::Soundex.
      - Removed HTTP::BrowserDetect.
      - Added URI::Escape

1.9.0  2005-08-17T23:35:49
      - Added new modules required for Bricolage 1.9.0:
        + Test::File::Contents.
        + DateTime
        + DateTime::TimeZone
        + Data::UUID

1.8.4
       - Added Term::ReadPassword.

1.8.3  2005-03-18T05:58:44
       - Added Test::MockModule.

1.8.2  2004-09-11T21:10:02
       - Now requires version 1.22 or later of DBD::Pg.

1.8.1  2004-07-08T05:39:64
       - Added Safe.pm.

1.8.0  2004-05-03T23:46:36
       - Updated for Bricolage 1.8.0, with several new modules added.
       - Removed Apache::ConfigFile, which is no longer needed by Bricolage.
       - Removed HTML::Template, which is now optional.
       - Removed Parse::RecDescent, which is not used directly by Bricolage,
         but by HTML::Template.

1.6.1  Tue Aug 26 18:08:42 2003
       - Switched to Module::Build.
       - Reorganized distribution layout to better reflect Perl standard.
       - Added POD tests.

1.6.0  Tue Apr 29 08:46:12 2003
       - Updated for Bricolage 1.6.0, with some new modules added.
       - Restored HTML::Mason to the list, now that 1.6.0 requires
         Mason 1.16 or later.

1.4.1  Thu Sep  5 01:30:41 2002
       - Removed HTML::Mason. It will be added back in once Bricolage has
         been ported to Mason 1.1x.

1.4.0  Wed Sep 04 04:33:32 2002
       - Updated for Bricolage 1.4.0, with some new modules added.
       - Updated the Bricolage URL.

1.3.4  Tue Jun 04 02:22:52 2002
       - Moved the Description above the contents so that they're easier to
         see when reading the documentation.

1.3.3  Tue Jun 04 02:13:14 2002
       - Added Cache::Mmap.
       - Added my name and email to Makefile.PL.

1.3.2  Wed May 08 15:52:09 2002
       - Added Apache::SizeLimit.
       - Require SOAP::Lite 0.55 to take advantage of the recent security fix.

1.3.1  Thu Feb 28 18:12:40 2002
       - Removed MD5 module.
       - Require Apache::Session 1.54.

1.3.0  Thu Feb 14 13:18:28 2002
       - Added modules required for SOAP support.

1.2.3  Fri Feb 08 21:14:03 2002
       - Added Changes file.
       - Added newly required Apache::ConfigFile module.
       - Added test.pl

1.2.2  Mon Feb 05 03:25:00 2002
       - Added Error.pm to the list of modules in the bundle, as it is
         required by Cache::Cache 0.99.

1.2.1  Fri Jan 13 01:33:00 2002
       - Fixed broken bundle - 1.2.0 didn't work at all!

1.2.0  Tue Jan 10 00:00:00 2002
       - Initial Public Release.


lib/Bundle/Bricolage.pm
package Bundle::Bricolage;

our $VERSION = '2.0.0';

1;
__END__

=head1 NAME

Bundle::Bricolage - Modules required to run Bricolage content management system.

=head1 SYNOPSIS

C<perl -MCPAN -e 'install Bundle::Bricolage'>

=head1 DESCRIPTION

The bundle provides an easy way to install all of the modules required by
Bricolage. Bricolage is a 100% pure Perl content-management and publishing
system which features intuitive and highly-configurable administration,
workflow, permissions, templating, server-neutral output, distribution, and
document management.

B<Note:> This bundle does not contain the modules that are optional in
Bricolage. To get those mosules, install
L<Bundle::BricolagePlus|Bundle::BricolagePlus>.

=head1 CONTENTS

Storable - Persistency for perl data structures

Time::HiRes - High resolution ualarm, usleep, and gettimeofday

Unix::Syslog - Perl interface to the UNIX syslog(3) calls

Net::Cmd - Network Command class (as used by FTP, SMTP etc)

Devel::Symdump - Dump symbol names or the symbol table

DBI 1.18 - Database independent interface for Perl

Error - Error/exception handling in an OO-ish way

Cache::Cache - The Cache interface

Cache::Mmap - Shared data cache using memory mapped files

Digest::MD5 - Perl interface to the MD5 Algorithm

Digest::SHA1 2.01 - Perl interface to the SHA-1 Algorithm

URI - Uniform Resource Identifiers (absolute and relative)

HTML::Tagset - Data tables useful in parsing HTML

HTML::Parser - HTML parser class

MIME::Base64 - Encoding and decoding of base64 strings

MIME::Tools - modules for parsing (and creating!) MIME entities

Mail::Address - Parse mail addresses

XML::Writer - Perl extension for writing XML documents

LWP - Library for WWW access in Perl

Image::Info - Extract meta information from image files

MLDBM - Store multi-level hash structure in single level tied hash

Params::Validate 0.57 - Validate method/function parameters

Exception::Class 1.12 - Perl Exceptions Base Class

Class::Container 0.09 - Glues object frameworks together transparently

mod_perl 1.30 - Embed a Perl interpreter in the Apache HTTP server

Apache::Request 1.0 - Generate compiler and linker flags for libapreq

HTML::Mason 1.23 - High-Performance, Dynamic Web Site Authoring System

DBD::Pg 1.22 - PostgreSQL database driver for the DBI module

DB_File - Perl5 access to Berkeley DB version 1.x

Apache::Session 1.54 - A persistence framework for session data

Test::Harness 2.03 - Run perl standard test scripts with statistics

Test::Simple - Basic utilities for writing tests

Test::MockModule 0.04 - Override subroutines in a module for unit testing

Test::File::Contents 0.02 - Test routines for examining the contents of files

Text::Balanced - Extract delimited text sequences from strings

XML::Parser 2.34 - A Perl module for parsing XML documents

XML::Simple - Easy API to read/write XML (esp config files)

IO::Stringy - I/O on in-core objects like strings and arrays

SOAP::Lite 0.55 - Client and server side SOAP implementation

File::Temp - Return Name and Handle of a Temporary File Safely

Text::LevenshteinXS - XS implementation of the Levenshtein edit distance

Locale::Maketext - Framework for Localization in Perl

Test::Class 0.04 - xUnit/JUnit style Test Suite System

Params::CallbackRequest 1.16 - Functional and object-oriented callback architecture

MasonX::Interp::WithCallbacks 1.10 - Mason callback support via Params::CallbackRequest

Safe 2.09 - Compile and execute code in restricted compartments

DateTime 0.21 - Date and time objects

DateTime::TimeZone 0.2601 - Time zone object base class and factory

Term::ReadPassword - Prompt for passwords without echoing to the terminal

Data::UUID - Generate globally/universally unique identifiers (GUIDs/UUIDs)

List::Util - A selection of general-utility list subroutines

List::MoreUtils - Provides the stuff missing in List::Util

Text::Diff - Perform diffs on files and record sets

Text::Diff::HTML 0.02 - XHTML format for Text::Diff::Unified

Text::WordDiff - Track changes between documents

URI::Escape - Escape and unescape unsafe URI characters

Scalar::Util - A selection of general-utility scalar subroutines

=head1 AUTHOR

David Wheeler <david@kineticode.com>

=head1 SEE ALSO

The L<Bricolage home page|http://bricolagecms.org/>.

See L<Bundle::BricolagePlus|Bundle::BricolagePlus> for modules that are
optional in Bricolage.

=head1 COPYRIGHT AND LICENSE

Copyright (c) 2002-2008, David E. Wheeler. Some Rights Reserved.

This module is free software; you can redistribute it and/or modify it under
the same terms as Perl itself.

=cut
